python/packages/nisar/products/writers/BaseL2WriterSingleInput.py

```diff
@@ -1996,6 +2041,105 @@ def geocode_metadata_group(self,
         input_raster_obj = isce3.io.Raster(
             input_temp.name, raster_list=input_raster_list)
 
+        geocode_kwargs = {}
+        if (lines == 1 or samples == 1):
+            geocode_kwargs['data_interpolator'] = 'nearest'
+
+        elif (lines < 5 or samples < 5):
+            geocode_kwargs['data_interpolator'] = 'bilinear'
```

Comment:
Let's add an optional argument to this function that lets the user decide which data interpolator to use.

If not specified, we'll fall back to this logic. (Please document the logic used here in the docstring.)